components/Editor.js

```diff
@@ -293,68 +299,66 @@ class Editor extends React.Component {
-              }
-              list={LANGUAGES}
-              onChange={this.updateLanguage}
-            />
-            <BackgroundSelect
-              onChange={this.updateBackground}
-              mode={backgroundMode}
-              color={backgroundColor}
-              image={backgroundImage}
-              aspectRatio={aspectRatio}
-            />
-            <Settings
-              {...config}
+      <div className="editor">
+        <Toolbar>
+          <Themes key={theme} updateTheme={this.updateTheme} theme={theme} />
+          <Dropdown
+            icon={languageIcon}
+            selected={
+              LANGUAGE_NAME_HASH[language] ||
+              LANGUAGE_MIME_HASH[language] ||
+              LANGUAGE_MODE_HASH[language] ||
+              LANGUAGE_MODE_HASH[DEFAULT_LANGUAGE]
+            }
+            list={LANGUAGES}
+            onChange={this.updateLanguage}
+          />
+          <BackgroundSelect
+            onChange={this.updateBackground}
+            mode={backgroundMode}
+            color={backgroundColor}
+            image={backgroundImage}
+            aspectRatio={aspectRatio}
+          />
+          <Settings
+            {...config}
+            onChange={this.updateSetting}
+            resetDefaultSettings={this.resetDefaultSettings}
+            format={this.format}
+            applyPreset={this.applyPreset}
+            getCarbonImage={this.getCarbonImage}
+          />
+          <div className="buttons">
+            <TweetButton onClick={this.upload} />
+            <ExportMenu
               onChange={this.updateSetting}
-              resetDefaultSettings={this.resetDefaultSettings}
-              format={this.format}
-              applyPreset={this.applyPreset}
-              getCarbonImage={this.getCarbonImage}
+              export={this.export}
+              exportSize={exportSize}
+              disablePNG={this.disablePNG}
             />
-            <div className="buttons">
-              <TweetButton onClick={this.upload} />
-              <ExportMenu
-                onChange={this.updateSetting}
-                export={this.export}
-                exportSize={exportSize}
-                disablePNG={this.disablePNG}
-              />
-            </div>
-          </Toolbar>
-
-          <Dropzone accept="image/*, text/*, application/*" onDrop={this.onDrop}>
-            {({ canDrop }) => (
-              <Overlay
-                isOver={canDrop}
-                title={`Drop your file here to import ${canDrop ? '✋' : '✊'}`}
+          </div>
+        </Toolbar>
+
+        <Dropzone accept="image/*, text/*, application/*" onDrop={this.onDrop}>
+          {({ canDrop }) => (
+            <Overlay
+              isOver={canDrop}
+              title={`Drop your file here to import ${canDrop ? '✋' : '✊'}`}
+            >
+              {/*key ensures Carbon's internal language state is updated when it's changed by Dropdown*/}
```

Comment:
@raboid I think we can remove this `key` now since the `Carbon` component doesn't have internal state

Reply:
What do you think?

Reply:
Makes sense to me!